source/slang/slang-ir-metal-legalize.cpp

```diff
@@ -671,10 +681,66 @@ namespace Slang
+                            groupThreadId = groupThreadIdBuilder.emitParamAtHead(getGroupThreadIdType(groupThreadIdBuilder));
+                            groupThreadIdBuilder.addNameHintDecoration(groupThreadId, groupThreadIDString);
+                            
+                            // Since "sv_groupindex" will be translated out to a global var and no longer be considered a system value
+                            // we can reuse its layout and semantic info
+                            Index foundRequiredDecorations = 0;
+                            IRLayoutDecoration* layoutDecoration = nullptr;
+                            UInt semanticIndex = 0;
+                            for (auto decoration : param->getDecorations())
+                            {
+                                if (auto layoutDecorationTmp = as<IRLayoutDecoration>(decoration))
+                                {
+                                    layoutDecoration = layoutDecorationTmp;
+                                    foundRequiredDecorations++;
+                                }
+                                else if (auto semanticDecoration = as<IRSemanticDecoration>(decoration))
+                                {
+                                    semanticIndex = semanticDecoration->getSemanticIndex();
+                                    groupThreadIdBuilder.addSemanticDecoration(groupThreadId, groupThreadIDString, (int)semanticIndex);
+                                    foundRequiredDecorations++;
+                                }
+                                if (foundRequiredDecorations >= 2)
+                                    break;
+                            }
+                            SLANG_ASSERT(layoutDecoration);
+                            layoutDecoration->removeFromParent();
+                            layoutDecoration->insertAtStart(groupThreadId);
+                            systemValWorkItems.add({ groupThreadId, groupThreadIDString, semanticIndex });
+                        }
+                    }
+
+                    IRBuilder svBuilder(builder.getModule());
+                    auto uintType = builder.getUIntType();
+                    auto newGroupIndex = svBuilder.createGlobalVar(uintType);
+                    svBuilder.addNameHintDecoration(newGroupIndex, UnownedStringSlice("sv_groupindex"));
+
+                    svBuilder.setInsertBefore(entryPoint.entryPointFunc->getFirstOrdinaryInst());
+                    auto computeExtent = emitCalcGroupExtents(svBuilder, entryPoint.entryPointFunc, builder.getVectorType(uintType, builder.getIntValue(builder.getIntType(), 3)));
+                    auto groupIndexCalc = emitCalcGroupThreadIndex(svBuilder, groupThreadId, computeExtent);
+                    svBuilder.emitStore(newGroupIndex, groupIndexCalc);
+
+                    param->replaceUsesWith(newGroupIndex);
```

Comment:
What type does `param` have? Isn't it a `int` instead of `int*`? If so we should just replace it with `groupIndexCalc`? Why do we need to replace a `param` which is local to the entrypoint, with a global variable? We should just insert the computation of the group index within the entrypoint and use the SSA value of the index as the param?

Reply:
I see your point, I did not need to make a globalVar